File: fibM.py
table={}
a=[]
def fibM(n):
    if n==0 or n==1:
        return n;
    else:
        if n not in table:
            
            table[n]=fibM(n-1)+fibM(n-2)
            
        return table[n]

n=int(input("enter the number of fib series: "))
for i in range(n):
    a.append(fibM(i))
print(a)
    

File: fibM.c
#include <stdio.h>
long int table[100] = {0};

long int FibM(int n) {
    if (n == 0 || n == 1) {
        return n;
    } else if (table[n] == 0) {
        table[n] = FibM(n - 1) + FibM(n - 2);
    }
    return table[n];
}

int main() {
    long int result = FibM(5);
    printf("Fib series num: %ld ", result);

    return 0;
}
